src/utilities/codemirror/vml-linter.ts

```diff
@@ -0,0 +1,288 @@
+import { syntaxTree } from '@codemirror/language';
+import { linter, type Diagnostic } from '@codemirror/lint';
+import type { Extension } from '@codemirror/state';
+import type { SyntaxNode, Tree } from '@lezer/common';
+import type { CommandDictionary, FswCommand, FswCommandArgument } from '@nasa-jpl/aerie-ampcs';
+import type { EditorView } from 'codemirror';
+import { closest } from 'fastest-levenshtein';
+import { VmlLanguage } from './vml';
+import {
+  TOKEN_DOUBLE_CONST,
+  TOKEN_ERROR,
+  TOKEN_HEX_CONST,
+  TOKEN_INT_CONST,
+  TOKEN_STRING_CONST,
+  TOKEN_UINT_CONST,
+} from './vml-constants';
+
+/**
+ * Limitations
+ *
+ * * Variables aren't checked, defer to external engine to determine if they exist when referenced
+ */
+
+// Absolute time tags may appear in functions beginning with SEQUENCE or ABSOLUTE_SEQUENCE
+// Functions beginning with BLOCK or RELATIVE_SEQUENCE may have only relative time tags.
+
+// Limit how many grammar problems are annotated
+const MAX_PARSER_ERRORS = 100;
+
+export function vmlLinter(commandDictionary: CommandDictionary | null = null): Extension {
+  return linter(view => {
+    const diagnostics: Diagnostic[] = [];
+    const tree = syntaxTree(view.state);
+    diagnostics.push(...validateParserErrors(tree));
+    if (!commandDictionary) {
+      return diagnostics;
+    }
+
+    const sequence = view.state.sliceDoc();
+    const parsed = VmlLanguage.parser.parse(sequence);
+
+    diagnostics.push(...validateCommands(commandDictionary, sequence, parsed));
+
+    return diagnostics;
+  });
+}
```

Comment:
Talked about using the `inputLinter` from the extension-point.ts as it is very similar

Reply:
The languages will have different checks and different sets of dictionaries, lets defer merging until we decide how to implement constant and global definitions